plugin/keyring_vault/vault_credentials_parser.cc

```diff
@@ -72,15 +76,41 @@ namespace keyring
   {
     reset_vault_credentials(&vault_credentials_in_progress);
 
-    std::ifstream credentials_file(file_url.c_str());
-    if (!credentials_file)
+    File_io file_io(logger);
+    File file = file_io.open(PSI_NOT_INSTRUMENTED,
+                             file_url.c_str(),
+                             O_RDONLY, MYF(MY_WME));
+
+    if (file < 0 || file_io.seek(file, 0, MY_SEEK_END, MYF(MY_WME)) == MY_FILEPOS_ERROR)
     {
       logger->log(MY_ERROR_LEVEL, "Could not open file with credentials.");
       return true;
     }
+    my_off_t file_size = file_io.tell(file, MYF(MY_WME));
+    if (file_size == 0)
+    {
+      logger->log(MY_ERROR_LEVEL, "Empty file with credentials.");
+      return true;
+    }
+    if (file_size > max_file_size ||
+        file_io.seek(file, 0, MY_SEEK_SET, MYF(MY_WME)) == MY_FILEPOS_ERROR)
+    {
+      logger->log(MY_ERROR_LEVEL, "Could not read file with credentials.");
+      return true;
+    }
+    boost::movelib::unique_ptr<uchar[]> buffer(new uchar[file_size]);
+    file_io.read(file, buffer.get(), file_size, MYF(MY_WME));
+    Secure_string file_content(reinterpret_cast<char*>(buffer.get()), file_size);
```

Comment:
Need to close the file after reading its content.

Reply:
Done